### PR TITLE
Buffs chem dispenser recharging

### PR DESCRIPTION
**Pull request by QualityVan**
## About The Pull Request
Triples the passive rate a chem dispenser recharges energy, from one point per eight seconds to three.
Adds the ability to take the battery out of chem dispensers, to replace or recharge faster.
Also fixes one misnamed icon.

## Why It's Good For The Game
Right now, you get one point every eight seconds, or 60 points every 8 minutes. To make a full pill bottle, you need 240 points. That's 32 minutes for one bottle. Speeding that up means it's something that can be attended to through the round, instead of just burning out of juice once and then being done.

## Changelog
:cl:
add: You can take the battery out of chem dispensers with a screwdriver and a crowbar.
balance: Chem dispensers recharge faster
fix: Orange vending machines will show their panel when screwdrivered
/:cl:

### CODE REVIEW

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 1]
```diff
@@ -132,8 +135,8 @@
 /obj/machinery/chem_dispenser/ui_data(mob/user)
 	. = list()
 	.["amount"] = amount
-	.["energy"] = cell.charge ? cell.charge * powerefficiency : "0" //To prevent NaN in the UI.
-	.["maxEnergy"] = cell.maxcharge * powerefficiency
+	.["energy"] = cell ? cell.charge * powerefficiency : "0" //To prevent NaN in the UI.
+	.["maxEnergy"] = cell ? cell.maxcharge * powerefficiency : "0"
```
Comment (TiviPlus) [inline code/modules/reagents/machinery/chem_dispenser.dm:139]: Why is this not just a ?.

You also have this being sent as a string and float

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 2]
```diff
@@ -288,21 +291,65 @@
...
+			return
+
+		else if(I.is_open_container())
+			if(!user.transferItemToLoc(I, src))
+				return
+
+			beaker =  I
+			to_chat(user, "You set [I] on the machine.")
+			updateUsrDialog()
+
```
Comment (TiviPlus) [inline code/modules/reagents/machinery/chem_dispenser.dm:306]: missing return

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 3]
```diff
@@ -288,21 +291,65 @@
...
+		else if(istype(I, /obj/item/reagent_containers/glass))
+			to_chat(user, "Take the lid off [I] first.")
+			return
+
+		to_chat(user, "The machine can't dispense into that.")
 		return
```
Comment (TiviPlus) [inline code/modules/reagents/machinery/chem_dispenser.dm:312]: dont need elseifs if you return

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 4]
```diff
@@ -288,21 +291,65 @@
...
+	return TRUE
 
+/obj/machinery/chem_dispenser/update_icon()
+	overlays.Cut()
+	if(!CHECK_BITFIELD(machine_stat, PANEL_OPEN))
+		return
+	else if(cell)
+		overlays += image(icon, "[initial(icon_state)]_open")
+	else
+		overlays += image(icon, "[initial(icon_state)]_nobat")
```
Comment (TiviPlus) [inline code/modules/reagents/machinery/chem_dispenser.dm:352]: use update_overlays

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/reagents/machinery/chem_dispenser.dm (hunk 1 above)
Comment (QualityVan) [reply code/modules/reagents/machinery/chem_dispenser.dm:139]: It needs to be 0 specifically for display stuff and ?. would give back null instead, didn't look too deeply into the ui it creates so I'm not certain how it's handling type.

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 5]
```diff
@@ -288,21 +291,66 @@
...
+		if(istype(I, /obj/item/reagent_containers/glass))
+			to_chat(user, "Take the lid off [I] first.")
+			return
+
+		to_chat(user, "The machine can't dispense into that.")
 		return
 
-	else if(istype(I, /obj/item/reagent_containers) && I.is_open_container())
+	else if(istype(I, /obj/item/cell))
+		if(!CHECK_BITFIELD(machine_stat, PANEL_OPEN))
```
Comment (TiviPlus) [inline code/modules/reagents/machinery/chem_dispenser.dm:316]:
```suggestion
	if(istype(I, /obj/item/cell))
		if(!CHECK_BITFIELD(machine_stat, PANEL_OPEN))
```

**Review by Kuro020 — APPROVED** [review]

**Comment by lbnesquik** [comment]
Oh yeah. Fuck chemistry, so making this more desirable is good.

**Pull request merged**